src/backend/distributed/utils/resource_lock.c

```diff
@@ -1196,3 +1245,169 @@ AcquireDistributedLockOnRelations(List *relationIdList, LOCKMODE lockMode)
 		SendCommandToWorker(nodeName, nodePort, lockCommand);
```

Comment:
Ah, missed that. It is probably more accurate to use `SendMetadataCommandListToWorkerInCoordinatedTransaction()`. It internally uses `METADATA CONNECTION` which is useful to prevent edge cases in tx blocks. We should use metadata connection because we are acting on the distributed shell tables

```C

SendMetadataCommandListToWorkerInCoordinatedTransaction(list_make1(lockCommand))
```

One additional test could be the following, which sometimes might fail without the fix:
```SQL
BEGIN;
create table tb(a int);
select create_distributed_table('tb', 'a');
lock  tb in access share mode ;
```